internal/dataplane/parser/parser.go

```diff
@@ -846,9 +847,18 @@ func listProtocols(svc *corev1.Service) map[corev1.Protocol]bool {
 func targetsForEndpoints(endpoints []util.Endpoint) []kongstate.Target {
 	targets := []kongstate.Target{}
 	for _, endpoint := range endpoints {
+		addr := endpoint.Address
+		parsed := net.ParseIP(endpoint.Address)
+		if parsed != nil {
+			if parsed.To4() == nil {
+				// if we have an IPv6 endpoint, we need to surround it with brackets, else the port concat after this will
+				// treat the port as part of the address
```

Comment:
```suggestion
				// If we have an IPv6 endpoint, we need to surround it with brackets, else the port concat after this will
				// treat the port as part of the address.
```